soc/nuvoton/npcm/common/pinctrl_soc.h

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2024 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_NUVOTON_NPCM_COMMON_PINCTRL_SOC_H_
+#define ZEPHYR_SOC_NUVOTON_NPCM_COMMON_PINCTRL_SOC_H_
+
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/types.h>
+#include <zephyr/dt-bindings/pinctrl/npcm-pinctrl.h>
+
+/**
+ * @brief NPCM peripheral pull-up/down configuration structure
+ *
+ * Used to indicate the peripheral device's corresponding group/bit for
+ * pull-up/down configuration.
+ */
+struct npcm_periph_pupd {
+	/** Properties used for io-pad. */
+	uint16_t io_bias_type: 2;
+	/** Reserved field. */
+	uint16_t reserved: 14;
+} __packed;
```

Comment:
I think packing is needed. Without packing, it adds one alignment byte and makes the address to access the union `cfg` in `struct npcm_pinctrl` unaligned. This would cause a compiler warning `-Waddress-of-packed-member`.